internal/provider/resource_aws_cloud_provider_credential.go

```diff
@@ -0,0 +1,148 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceCloudProviderCredentialAws() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudProviderCredentialAwsCreate,
+		ReadContext:   resourceCloudProviderCredentialRead,
+		UpdateContext: resourceCloudProviderCredentialAwsUpdate,
+		DeleteContext: resourceCloudProviderCredentialDelete,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Description of the Cloud Provider Credentials.",
+			},
+			"aws_access_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
This should be required - or is it optional because of the fallback to the env var?

Reply:
You are right, it should be required. I had it set as optional because at the beginning I created a single resource for both AWS and Google but decided to split it into 2.